### DISCUSSION TITLE
Several Variables in SpecifiedSmoothCircleIC

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#27058] [Q&A General] [answered]
Hello everyone, I was wondering if it is possible to add several variables in the SpecifiedSmoothCircleIC (because I am receiving an error saying that the initial condition does not exist), for example:

  [grain_1_IC]
    type = SpecifiedSmoothCircleIC
    variable = 'gr0 gr1 gr2'
    x_positions = '100.0 150.0 200.0'
    y_positions = '50.0 50.0 50.0'
    z_positions = '0.0 0.0 0.0'
    radii = '20.0 20.0 20.0'
    invalue = 1.0
    outvalue = 0.0
    int_width = 5.0
    3D_spheres = false
  []

So the idea is to initialize several new grains in a geometry and want each of them to be with a different euler angle and independent from each other.

Thanks in advance!

**Comment by GiudGiud** [comment]
Hello

I think this is definitely doable programmatically. Input file wise not sure,

What error are you getting exactly ?

Guillaume

**Comment by fervaldies** [comment]
It says:

*** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".

Variable 'gr0 gr1 gr2' requested in initial condition 'grain_1_IC' does not exist.

**Reply by GiudGiud** [reply]
I think you need to use the related action that creates variables as well as the IC. (I think they do that I ll check tomorrow)

Check out some example inputs for the ICs in phase field

**Reply by jiangwen84** [reply, marked as answer]
As @GiudGiud mentioned, it is not supported for multiple variables. Each IC can only assign the value to a single variable. You can use three IC blocks to assign the ICs to gr0, gr1 and gr2 separately.
